feat: EPI-1128: Build workflow to record a medication as Given

huytranbes opened the pull request:
### Changes

_Add a brief description of the changes in this PR to help give the reviewer context._

### Deploys

- [ ] **Deploy to Tamanu Internal** <!-- #deploy -->

### Tests

- [ ] **Run E2E Tests** <!-- #e2e -->

### Remember to...

- ...write or update tests
- ...add UI screenshots and **testing notes** to the Linear issue
- ...add any **manual upgrade steps** to the Linear issue
- ...update the [config reference](https://beyond-essential.slab.com/posts/reference-config-file-0c70ukly), [settings reference](https://beyond-essential.slab.com/posts/reference-settings-0blw1x2q), or any [relevant runbook(s)](https://beyond-essential.slab.com/topics/runbooks-bs04ml6c)
- ...call out additions or changes to **config files** for the deployment team to take note of

<!-- Thank you! -->

passcod commented on packages/database/src/migrations/1742897789276-addMedicationAdministrationRecordDoseTable.ts:
```diff
@@ -0,0 +1,47 @@
+    id: {
+      type: DataTypes.UUID,
+      allowNull: false,
+      primaryKey: true,
```
```suggestion
      primaryKey: true,
      defaultValue: Sequelize.fn('gen_random_uuid'),
```

passcod commented on packages/database/src/migrations/1742897789276-addMedicationAdministrationRecordDoseTable.ts:
```diff
@@ -0,0 +1,47 @@
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    created_at: {
+      type: DataTypes.DATE,
+      allowNull: false,
+    },
+    updated_at: {
+      type: DataTypes.DATE,
+      allowNull: false,
+    },
```
```suggestion
    created_at: {
      type: DataTypes.DATE,
      allowNull: false,
      defaultValue: Sequelize.fn('now'),
    },
    updated_at: {
      type: DataTypes.DATE,
      allowNull: false,
      defaultValue: Sequelize.fn('now'),
    },
```

passcod commented on packages/facility-server/app/routes/apiv1/encounter.js:
```diff
@@ -203,7 +203,13 @@ encounterRelations.get(
+            'reasonNotGiven', 
+            {
+              association: 'doses',
+              order: [['createdAt', 'ASC']]  // Order doses by oldest first
```
this should use `givenTime` I think, `createdAt` is affected by sync so not reliable

passcod commented on packages/facility-server/app/routes/apiv1/medication.js:
```diff
@@ -458,6 +458,89 @@ medication.get(
+  }),
+});
+medication.put(
+  '/mar/:id/given',
```
is MAR standard or something we've used elsewhere? would prefer the meaning be spelled out if not

passcod replied:
(in the route and also in exported functions elsewhere; local variables can remain short though something like `const record = await...` might be clearer.)

passcod reviewed (approved):
a few details but otherwith 👍🏻

huytranbes committed:
feat: EPI-1128: Build workflow to record a medication as Given

huytranbes committed:
feat(medication): Add DRUG_UNIT_SHORT_LABELS to enumRegistry and update MarStatus component

- Introduced DRUG_UNIT_SHORT_LABELS to provide short representations for drug units in the enumRegistry.
- Updated MarStatus component to utilize TranslatedEnum for displaying drug units, enhancing clarity in medication dosage presentation.

huytranbes committed:
feat(medication): Enhance MarStatus and StatusPopper components with future and past medication handling

- Added `isFuture` and `isPast` props to MarStatus and StatusPopper components to improve medication status management.
- Updated dose amount initialization and validation logic in GivenScreen to accommodate new props.
- Enhanced the handling of time-related logic for medication administration based on the new props, improving user experience and accuracy in medication tracking.
